### ISSUE TITLE
JDK16: NullPointException Cannot invoke "Object.getClass()" because "cause" is null

### ISSUE THREAD

**Issue by krasosu** [tkaczmarzyk/specification-arg-resolver#122]
In JDK16 there is a NullPointerException related to @Join annotations. I spent days looking for the error. The same code works successfully with JDK8.

```
  @GetMapping(value = "/search")
  public ResponseEntity<List<Person>> search(
          @Join(path = "skills", alias = "s")
          @Spec(path = "s.type", params = "skilltype", spec = Equal.class) Specification<Person> spec) {

    return ResponseEntity.ok().body(this.personService.getPersonBySpecification(spec));

  }
```

[stackoverflow-67840388](https://stackoverflow.com/questions/67840388/spring-boot-jpa-specification-arg-resolver-joined-table/67843493#67843493)

Unit Tests fail with 16JDK with the following error message:
```
Caused by: org.springframework.cglib.core.CodeGenerationException: java.lang.reflect.InaccessibleObjectException-->Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @5fb2de77

```

Since Java 9 build 148, it is no longer possible to do setAccessible on any public Java runtime API class (except sun.misc.Unsafe as special case). This affects ClassLoader#defineClass (protected method).

**Comment by muety** [comment]
Same problem here. Any plans to solve this?

**Comment by SimonBalling** [comment]
+1

**Comment by andrejs-sinkevics** [comment]
+1

**Comment by kekcogen** [comment]
++

**Comment by stlegion** [comment]
Same probkem. Migrating from Java 8 to 17 and now it's impossible to use Join annotation.

**Comment by s-frei** [comment]
Apparently it seems like I fixed it with #129
Please reach out to me if this approach isn't correct 😉

**Comment by tkaczmarzyk** [comment]
Fixed with #129 and released in v2.6.3. Sorry for the delay -- I got involved in family life and didn't notice time passing by ;) Thanks for reporting and all contributions!
